platforms/core-runtime/logging/src/test/groovy/org/gradle/internal/deprecation/DeprecationMessagesTest.groovy

```diff
@@ -44,9 +42,8 @@ class DeprecationMessagesTest extends Specification {
     def setup() {
         def diagnosticsFactory = new NoOpProblemDiagnosticsFactory()
 
-        def buildOperationProgressEventEmitter = Stub(BuildOperationProgressEventEmitter)
-        def problemEmitter = Stub(ProblemEmitter)
-        DeprecationLogger.init(diagnosticsFactory, WarningMode.All, buildOperationProgressEventEmitter, new DefaultProblems(problemEmitter))
+        def mock = Mock(BuildOperationProgressEventEmitter)
```

Comment:
💅 I think it's better to create it with `Stub` if you're not interested in verifying interactions with a test double.